job_crafter.py

```diff
@@ -102,75 +123,97 @@ def make_jobs(self):
 
         self.job["device_type"] = self.board['device_type']
 
-        if self.options['default_notify']:
+        if self.cfg['default_notify']:
             self.job["notify"] = self.board.get("notify", [])
         else:
-            self.job["notify"] = self.options["notify"]
-        print("notify recipients: Overridden")
+            self.job["notify"] = self.cfg['notify']
+
+        logging.info("Notifications recipients: %s" % ", ".join(self.job['notify']))
 
         # Define which test to run
         tests = []
-        if self.options['tests']:
-            tests = [next(iter([e for e in self.board['tests'] if e['name'] == t]), {'name': t}) for t in self.options['tests']]
+        if 'tests' in self.cfg:
+            # WTF?
+            tests = [next(iter([e for e in self.board['tests'] if e['name'] == t]), {'name': t}) for t in self.cfg['tests']]
         else:
             tests = self.board.get("tests", [])
         for test in tests:
-            if self.options['kernel']:
+            logging.info("Configuring test: %s" % test['name'])
+            if 'kernel' in self.cfg:
                 defconfigs = ['custom_kernel']
+
                 # If we use a custom kernel
-                if self.options["dtb"]:
-                    dt_path = os.path.abspath(self.options["dtb"])
+                if 'dtb' in self.cfg:
+                    dt_path = os.path.abspath(self.cfg['dtb'])
                 else:
-                    dt_path = os.path.abspath(os.path.join(self.options["dtb_folder"],
-                        self.board['dt'] + '.dtb'))
+                    dt_path = os.path.abspath(os.path.join(self.cfg['dtb_folder'],
+                                                           self.board['dt'] + '.dtb'))
+
                 data = {
-                        'kernel': self.options['kernel'],
+                        'kernel': self.cfg['kernel'],
                         'dtb': dt_path,
-                        'modules': self.options['modules'],
+                        'modules': self.cfg['modules'],
                         }
+
             else:
-                defconfigs = self.options['defconfigs'] or test.get('defconfigs', self.board['defconfigs'])
+                if 'defconfigs' in self.cfg:
+                    defconfigs = self.cfg['defconfigs']
+                else:
+                    defconfigs = test.get('defconfigs', self.board['defconfigs'])
+
             for defconfig in defconfigs:
-                if not self.options['kernel']:
-                    # No custom kernel, go fetch artifacts on remote locations
-                    data = (ArtifactsFinder("http://lava.free-electrons.com/downloads/builds/",
-                            **self.options).crawl(self.board, defconfig) or
-                            ArtifactsFinder("https://storage.kernelci.org/",
-                            **self.options).crawl(self.board, defconfig))
```

Comment:
Removing that breaks sending custom artifacts.  
Since the defconfig will be turned in "custom_defconfig", it's something that can't be found, and we'll end up with the `IOError`.  
I really think we should not try to fetch artifacts from remote location (except maybe the rootfs) when we supply custom kernel.